internal/impl/devbox.go

```diff
@@ -125,7 +125,10 @@ func Open(path string, writer io.Writer) (*Devbox, error) {
 	if err != nil {
 		return nil, err
 	}
-	box.pluginManager.ApplyOptions(plugin.WithLockfile(lock))
+	box.pluginManager.ApplyOptions(
+		plugin.WithDevbox(box),
+		plugin.WithLockfile(lock),
```

Comment:
Is providing a Devbox and/or lockfile optional?

Reply:
@gcurtis not really, but due to circular dependency needs to be added after construction